code/modules/mining/machine_processing.dm

```diff
@@ -5,6 +5,33 @@
 /obj/machinery/mineral
 	var/input_dir = NORTH
 	var/output_dir = SOUTH
+	var/turf/input_turf = null // The turf that the machines will be listening to for ores/items
+	var/needs_item_input = FALSE // whether or not the machine needs to pick up items, i.e. the ORM picking up ores
+	processing_flags = START_PROCESSING_MANUALLY
+	subsystem_type = /datum/controller/subsystem/processing/fastprocess
+
+/obj/machinery/mineral/Initialize(mapload)
+	. = ..()
+	if(needs_item_input)
+		register_input_turf()
+
+/obj/machinery/mineral/Destroy()
+	. = ..()
+	if(input_turf)
+		unregister_input_turf()

```

Comment:
On deletion things will unregister themselves clearly already.
You'll only need to care about things registered to the thing being deleted.

Reply:
I'm confused by what you mean here, can you re-phrase?

Reply:
destroy() will automatically handle you unregister to any events, but there might still be things listening to events of the qdel'd object

Reply:
Ah, so basically I don't even need this Destroy override, the signal should unregister itself when the machine is destroyed?

Reply:
Yep. When something is destroyed it will automatically unregister from everything.
And actually, to correct myself, it will also unregister properly things listening to it, so no need to worry about it.
You can remove this from `Destroy()` safely.